RandomMCPlayer.java

```diff
@@ -168,7 +170,192 @@ private int simPlay(int depthLimit) {
+			expectedValue += probOfSequence(hand, suitArrayList.indexOf(4)) * 30; // straight flush
+			expectedValue += probOfSuit(suitArrayList.indexOf(4)) * 5; // flush
+		}
+		
+		if (rankArrayList.contains(3)) {
+			expectedValue += probOfRank(rankArrayList.indexOf(3)) * 16; // fourOfKind
+			expectedValue += probOfRank(rankArrayList.indexOf(1)) * 10; // find card to complete 2 card rank for full-house 
+		}
+		else if (rankArrayList.contains(2)) {
+			if (Collections.frequency(rankArrayList, 2) == 2) { // check if there's two occurence of 2 
+				expectedValue += probOfRank(rankArrayList.indexOf(2), rankArrayList.lastIndexOf(2)) * 10; // find card for 3 card rank of full-house
+			}
+			else {
+				expectedValue += probOfRank(rankArrayList.lastIndexOf(2)) * 6; // threeOfKind
+				expectedValue += probOfRank(rankArrayList.lastIndexOf(1)) * 3; // twoPair
+			}
+		}
+		else {
+			expectedValue += probOfRank(rankArrayList) * 1; // onePair
+		}
+
+		expectedValue += probOfSequence(hand, -1) * 12; // straight flush
+
+		return expectedValue;
+	}
+
+	float probOfRoyalFlush(Card[] hand, int suit) {
+		int rankToGet = 0;
+		boolean correctSequence = true;
+		int undealtRoyalCardCount = 0;
+
+		for (int i=0; i < 5; i++) {
+			if (hand[i] == null ) {
+				rankToGet = (i==4)? 0: (i+10);
+			}
+			else if (hand[i].getRank() != ((i==4)? 0: (i+10))){
+				correctSequence = false;
+			}
+		}
+
+		if (correctSequence) { // probability of missing rank with specified suit
+			for (int i=numPlays; i<simDeck.length; i++) {
+				Card card = simDeck[i];
+				if (card.getSuit() == suit && card.getRank() == rankToGet) undealtRoyalCardCount++;
+			}
+			return undealtRoyalCardCount;
+		}
+		else {
+			return 0;
+		}
+	}
 	
+	float probOfSequence(Card[] hand, int suit) {
+		int rankToGet = 0;
+		int [] updatedHandRanks = new int [5];
+		int eligibleCards = 0;
+
+		for (int i=0; i < 5; i++) {
+			if (hand[i] == null ) {
+				if (i == 0){
+					rankToGet = hand[1].getRank() - 1;
+					updatedHandRanks[i] = rankToGet;
+				}
+				else {
+					rankToGet = hand[i-1].getRank() + 1;
+					updatedHandRanks[i] = rankToGet;
+				}
+				updatedHandRanks[i] =  hand[i].getRank();
+			}
+		}
+
+		boolean correctSequence = true;
+		for (int i=0; i < 4; i++) {
+			if (updatedHandRanks[i] != hand[i+1].getRank() - 1) {
+				correctSequence = false;
+			}
+		}
+
+		if (correctSequence) {
+			for (int i=numPlays; i<simDeck.length; i++) {
+				Card card = simDeck[i];
+				if (card.getRank() == rankToGet && ( suit == -1 || card.getSuit() == suit)) {
+					eligibleCards++;
+				} 
+			}
+			return eligibleCards/ simDeck.length;
+		}
+		else {
+			return 0;
+		}
+	}
+
+	float probOfSuit(int suitIndex) {
+		int undealtSuitCount = 0;
+		for (int i=numPlays; i<simDeck.length; i++) {
+			Card card = simDeck[i];
+			if (card.getSuit() == suitIndex) undealtSuitCount++;
+		}
+		return (undealtSuitCount/simDeck.length);
+	}
```

Comment:
Should be dividing by `NUM_CARDS-numPlays` instead of `simDeck.length`.
(`simDeck.length` is always 52)
Note: this should be fixed for all probability functions.